app/middleware/actionLogging.js

```diff
@@ -57,32 +57,52 @@ function _getChangeKind(changeCode) {
   }
 }
 
-function _getDiff(oldVal, newVal) {
+/**
+ * Create change log for array and objects
+ * @param obj|array oldVal
+ * @param obj|array newVal
+ * @param string Action type received
+ * @return string Change log
+ */
+function _getDiff(oldVal, newVal, actionType) {
   const calcDiff = diff(oldVal, newVal);
-  return !calcDiff || !calcDiff.length ? 'No changes' :
-    calcDiff.map((change) => {
-      let kind = _getChangeKind(change.kind);
-      let location;
-
-      // If not an array change
-      if ('A' !== change.kind) {
-        location = change.path.join('.');
-      } else {
-        // If an array change
-        kind = _getChangeKind(change.item.kind);
-        location = `index ${change.index}`;
-      }
-
-      return `${kind} at ${location}`;
-    }).join("\n"); // eslint-disable-line quotes
+  if (!calcDiff || !calcDiff.length) {
+    return 'No changes';
+  }
+
+  let changeLog = '';
+  calcDiff.forEach((change) => {
```

Comment:
Instead of a forEach, lets keep this functional and use a reduce:

```
return calcDiff.reduce(
  (log, change) => {
    /* all the function stuff */
    return `${log}${$kind at ${location}\n`;
  }, ''
).trim();
```